pyuvdata/uvdata/uvh5.py

```diff
@@ -375,11 +374,17 @@ def close(self):
         """Close the file."""
         self.__header = None
         self.__datagrp = None
-        del self.header  # need to refresh these
+
+        try:
+            del self.header  # need to refresh these
+        except AttributeError:
+            pass
```

Comment:
The coverage issue on this line is a known problem -- it's caused by optimizations in the compiler. You should check that you have a test that goes down this branch and if you do, you can safely mark this with a `pragma: no cover`.